src/runtime/threadpool/builder.rs

```diff
@@ -330,14 +331,24 @@ impl Builder {
         // Get a handle to the clock for the runtime.
         let clock = self.clock.clone();
 
-        let pool = self.threadpool_builder
+        // Get the current trace dispatcher.
+        // TODO(eliza): when `tokio-trace-core` is stable enough to take a
+        // public API dependency, we should allow users to set a custom
+        // subscriber for the runtime.
+        let dispatch = dispatcher::get_default(Dispatch::clone);
+
+        let pool = self
+            .threadpool_builder
             .around_worker(move |w, enter| {
                 let index = w.id().to_usize();
+                let dispatch = dispatch.clone();
 
                 tokio_reactor::with_default(&reactor_handles[index], enter, |enter| {
                     clock::with_default(&clock, enter, |enter| {
                         timer::with_default(&timer_handles[index], enter, |_| {
-                            w.run();
+                            dispatcher::with_default(dispatch, || {
```

Comment:
Just noting, out of context, "dispatcher" sounds like a task dispatcher, not something about traces...

Reply:
@seanmonstar in f409c12c9d238c33a589b6f5348291dc2c29ec88, I changed the way the `Dispatch` type and the `dispatcher` module are imported so that they're all namespaced as `trace::dispatcher` --- hopefully that's clearer!

Reply:
I do agree w/ sean... `trace::dispatcher` helps some. IMO it would be better if it was `trace::subscriber`, but I also know why it isn't subscriber (`Dispatcher::new(subscriber)`).

I'm not going to make a request either way. I just wanted to provide these thoughts to @seanmonstar and @hawkw